Yoast/wordpress-seo
# UI library & dependencies release

## Context


* The `@yoast/ui-library` has changed in several places since its 3.0.0 release. We want to release the newest version.
* The `@yoast/ui-library` has several published dev-dependencies that have also had changes in the meantime. We want to release those newest versions.

## Summary


This PR can be summarized in the following changelog entry:

* Updates versions and changelogs of several packages, and updates their relationships.

## Relevant technical choices:

* The changelogs are generated based on the changelog entries written in pull requests. I focused on `@yoast/ui-library` and all internal dependencies of that package.
* Lerna looks for changes since the last Git tag. Because of this behaviour, I added changelogs manually first to make Lerna detect the changed packages since one of the 20.1-RCx tags.
  * Both `yoast-components` and `@yoast/analysis-report` are now configured as private packages, to prevent Lerna from trying to update them.
* The increment sizes (4x minor increment) are based on the copy in the changelog. None of the changelog items describe a backwards compatibility breaking change.
* While running into issues, I updated Lerna to try and solve some. I kept that update in, since it didn't break anything.
  * We only use Lerna for `$ yarn lint`, `$ yarn test`, and `$ yarn test --ci`.
  * Lerna by default now runs commands in a sorted order based on package dependencies. I've disabled that with the `--no-sort` flag, to revert to the original state for linting and testing.
* The `package.json` file in the root of the workspace also depends on our packages. Due to this, the resolved `node_modules` structure might end up as one big mess. So I've updated those as well to the latest versions.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

Because the plugin is always built with the code that lives inside the repo, the latest version was already using a version of the packages ahead of the last release. So to test this, **a smoke test will be sufficient**.

**To test the Lerna changes:**

* In the root of the repo run `$ yarn install`, then all these should pass and work as before:
  * Run `$ yarn lint`
  * Run `$ yarn test`
  * Run `$ yarn test --ci`

#### Relevant test scenarios


The changes touch the JavaScript packages. Those are (used for building) the visual elements of the pages in the plugin.

### Test instructions for QA when the code is in the RC





## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

* Scripts for workflows around building the plugin.
* The UI library and thus indirectly the parts of the plugin using it, like the Settings page.

## UI changes


While counter intuitive, **this PR should not change the UI of the plugin**.

## Other environments


## Documentation

* [ ] I have written documentation for this change. **Not yet, but will document the process (and steps) I've followed thus far.**

## Quality assurance


## Innovation
